holoviews/operation/datashader.py

```diff
@@ -148,32 +166,52 @@ def _process(self, element, key=None):
 
         cvs = ds.Canvas(plot_width=width, plot_height=height,
                         x_range=(xstart, xend), y_range=(ystart, yend))
+        return getattr(cvs, glyph)(data, x, y, self.p.aggregator)
 
-        if line:
-            return cvs.line(data, x, y, self.p.aggregator)
-        else:
-            return cvs.points(data, x, y, self.p.aggregator)
-            
-
-def uint32_to_uint8(img):
-    return np.flipud(img.view(dtype=np.uint8).reshape(img.shape + (4,)))
 
 
 class Shade(ElementOperation):
+    """
+    Shade applies a normalization function to the data and then
+    applies colormapping to an Image or NdOverlay of Images, returning
```

Comment:
Maybe explain what "the data" is?  I.e. is it an xarray, either 2D or if 3D then a stack of categories?